core-bundle/src/Security/Authentication/ContaoLoginAuthenticationListener.php

```diff
@@ -48,7 +48,7 @@ protected function requiresAuthentication(Request $request): bool
     {
         return $request->isMethod('POST')
             && $request->request->has('FORM_SUBMIT')
-            && preg_match('/^tl_login(_[0-9]+)?$/', $request->request->get('FORM_SUBMIT'));
+            && preg_match('/^tl_login(_[0-9]+)?$/', (string) $request->request->get('FORM_SUBMIT'));
```

Comment:
Is it really wise to just cast the value and ignore the error? There is a similar discussion in https://github.com/contao/contao/pull/6567#pullrequestreview-1750226004 where I suggested to throw an exception instead.

Reply:
The [issue](https://github.com/contao/contao/issues/6702) is that malicious login attempts flood the log with errors. Either we prevent that error and thus avoid flooding the log - or we don't. There is not really another way, is there?

Reply:
And in any case: in Symfony 5 `InputBag::get` can return `string|int|float|bool|null`, so casting to `(string)` makes sense anyway, does it not? (i.e. defensive programming)

Reply:
#6567 is a different case anyway. There you cannot cause the error by flooding the server with invalid requests. It is caused in the back end and presumably has a specific reproduction.

Reply:
The question is whether we should ignore hacking attempts or not. I find it more defensive to throw "bad request" exceptions in such cases than to just cast the value and hope that the application will handle the attack later on.

The "log flooding" is just a symptom. We should fix the root cause IMHO.

Reply:
```suggestion
            && \is_string($request->request->get('FORM_SUBMIT'));
            && preg_match('/^tl_login(_[0-9]+)?$/', $request->request->get('FORM_SUBMIT'));
```